packages/datetime/test/dateRangeInputTests.tsx

```diff
@@ -33,4 +33,35 @@ describe("<DateRangeInput>", () => {
         expect(inputRef.calledOnce).to.be.true;
         expect(inputRef.firstCall.args[0]).to.be.an.instanceOf(HTMLInputElement);
     });
+
+    it("invokes onChange when a day is selected or deselected in the picker", () => {
+        const onChange = sinon.spy();
+        const { root, getDayElement } = wrap(<DateRangeInput onChange={onChange} />);
+        root.setState({ isOpen: true });
+
+        // select days
+        getDayElement(22).simulate("click");
+        getDayElement(24).simulate("click");
+
+        // deselect days
+        getDayElement(22).simulate("click");
+        getDayElement(24).simulate("click");
+
+        expect(onChange.callCount).to.deep.equal(4);
```

Comment:
also verifying the internal state (including null states since those get convoluted quickly) in uncontrolled usage would be useful - with the standard `when uncontrolled` and `when controlled` blocks that we have in other tests